auth/src/main/java/com/firebase/ui/auth/ui/email/WelcomeBackPasswordPrompt.java

```diff
@@ -111,73 +107,39 @@ protected void onCreate(Bundle savedInstanceState) {
             @Override
             public void onChanged(@Nullable Resource<IdpResponse> resource) {
-                onAuthResult(resource);
+                onSignInOperation(resource);
             }
         });
     }
 
-    @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        // Forward activity results to the ViewModel
-        if (!mHandler.onActivityResult(requestCode, resultCode, data)) {
-            super.onActivityResult(requestCode, resultCode, data);
-        }
-    }
-
-    private void onAuthResult(@Nullable Resource<IdpResponse> resource) {
+    private void onSignInOperation(@Nullable Resource<IdpResponse> resource) {
         if (resource == null) {
             Log.w(TAG, "Got null resource, ignoring.");
             return;
         }
 
-        if (resource.getState() == State.LOADING) {
-            getDialogHolder().showLoadingDialog(R.string.fui_progress_dialog_signing_in);
-        }
-
-        if (resource.getState() == State.SUCCESS) {
-            getDialogHolder().dismissDialog();
-            Log.d(TAG, "onAuthResult:SUCCESS:" + resource.getValue());
-
-            finish(RESULT_OK, resource.getValue().toIntent());
-        }
-
-        if (resource.getState() == State.FAILURE) {
-            getDialogHolder().dismissDialog();
-
-            // TODO: Is this message what we want?
-            String message = resource.getException().getLocalizedMessage();
-            mPasswordLayout.setError(message);
-        }
-    }
-
-    private void onPendingResolution(@Nullable PendingResolution resolution) {
-        if (resolution == null) {
-            Log.e(TAG, "Got null resolution, can't do anything");
-            return;
-        }
-
-        try {
-            startIntentSenderForResult(resolution.getPendingIntent().getIntentSender(),
-                    resolution.getRequestCode(), null, 0, 0, 0);
-        } catch (IntentSender.SendIntentException e) {
-            Log.e(TAG, "Failed to send resolution.", e);
-
-            IdpResponse errorResponse = IdpResponse.fromError(
-                    new FirebaseUiException(ErrorCodes.UNKNOWN_ERROR, getString(R.string.fui_general_error), e));
-            finish(RESULT_OK, errorResponse.toIntent());
+        switch (resource.getState()) {
+            case LOADING:
+                getDialogHolder().showLoadingDialog(R.string.fui_progress_dialog_signing_in);
+                break;
+            case SUCCESS:
+                getDialogHolder().dismissDialog();
+
+                // This logic remains in the view since SmartLock is effectively a different
+                // 'screen' after the sign-in process.
+                FirebaseUser user = getAuthHelper().getCurrentUser();
+                startSaveCredentials(user, mHandler.getPendingPassword(), resource.getValue());
+                break;
+            case FAILURE:
+                // TODO: Is this message what we want?
```

Comment:
Don't we have an "incorrect password" string? We could see if the exception is an instance of the `FirebaseAuthInvalidCredentialsException` and use that string or else show the localized message. That's better than what we have. 😁

Reply:
We actually don't ... I added a TODO to make one since I don't want to add one here and get a red build for a non-translated string.